hoomd_organics/library/forcefields.py

```diff
@@ -187,3 +188,204 @@ def _create_forcefield(self):
                 periodic_dihedral.params[dih_type] = self.dihedrals[dih_type]
             forces.append(periodic_dihedral)
         return forces
+
+
+class TableForcefield:
+    """Load table potential forcefields."""
+
+    def __init__(
+        self,
+        pairs=None,
+        bonds=None,
+        angles=None,
+        dihedrals=None,
+        r_min=None,
+        r_cut=None,
+        exclusions=["bond", "1-3"],
+        nlist_buffer=0.40,
+    ):
+        self.pairs = pairs
+        self.bonds = bonds
+        self.angles = angles
+        self.dihedrals = dihedrals
+        self.r_min = r_min
+        self.r_cut = r_cut
+        self.exclusions = exclusions
+        self.nlist_buffer = nlist_buffer
+        self.bond_width, self.angle_width, self.dih_width = self._check_widths()
+        self.hoomd_forcefield = self._create_forcefield()
+
+    @classmethod
+    def from_files(
+        cls,
+        pairs=None,
+        bonds=None,
+        angles=None,
+        dihedrals=None,
+        exclusions=["bond", "1-3"],
+        nlist_buffer=0.40,
+    ):
+        """Create forcefield from text files containing tabulated forces."""
+        # Read pair files
+        pair_dict = dict()
+        r_min = set()
+        r_max = set()
+        if pairs:
+            for pair_type in pairs:
+                table = np.loadtxt(pairs[pair_type])
+                r = table[:, 0]
+                r_min.add(r[0])
+                r_max.add(r[-1])
+                pair_dict[pair_type] = dict()
+                pair_dict[pair_type]["U"] = table[:, 1]
+                pair_dict[pair_type]["F"] = table[:, 2]
+            if len(r_min) != len(r_max) != 1:
```

Comment:
Technically, we can handle pair tables with different r_mins and r_maxs if we set the rmax on a per-pair basis in `_create_hoomd_forcefield`